algorithms/src/Kokkos_Sort.hpp

```diff
@@ -634,6 +640,29 @@ sort(const ExecutionSpace& exec,
   bin_sort.sort(exec, view);
 }
 
+#if defined(KOKKOS_ENABLE_ONEDPL)
+template <class DataType, class... Properties>
+void sort(const Experimental::SYCL& space,
+          const Kokkos::View<DataType, Properties...>& view) {
+  using ViewType = Kokkos::View<DataType, Properties...>;
+  static_assert(
+      ViewType::rank == 1 &&
+          (std::is_same<typename ViewType::array_layout, LayoutRight>::value ||
+           std::is_same<typename ViewType::array_layout, LayoutLeft>::value),
+      "SYCL sort only supports contiguous 1D Views.");
```

Comment:
I think the message is misleading.
This is a restriction for `Kokkos::sort` not something specific from `SYCL`.  We get it transitively from calling `Experimental::{begin,end}(view)` in other specializations of the algorithm.